File: RollDice/src/main/java/com/ssaurel/dicer/MainActivity.java
package com.ssaurel.dicer;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.ImageView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

	public static final Random RANDOM = new Random();
	private Button rollDices;
	private ImageView imageView1, imageView2;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		rollDices = (Button) findViewById(R.id.rollDices);
		imageView1 = (ImageView) findViewById(R.id.imageView1);
		imageView2 = (ImageView) findViewById(R.id.imageView2);

		rollDices.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
				final Animation anim1 = AnimationUtils.loadAnimation(MainActivity.this, R.anim.shake);
				final Animation anim2 = AnimationUtils.loadAnimation(MainActivity.this, R.anim.shake);
				final Animation.AnimationListener animationListener = new Animation.AnimationListener() {
					@Override
					public void onAnimationStart(Animation animation) {
					}

					@Override
					public void onAnimationEnd(Animation animation) {
						int value = randomDiceValue();
						int res = getResources().getIdentifier("dice_" + value, "drawable", "com.ssaurel.dicer");

						if (animation == anim1) {
							imageView1.setImageResource(res);
						} else if (animation == anim2) {
							imageView2.setImageResource(res);
						}
					}

					@Override
					public void onAnimationRepeat(Animation animation) {

					}
				};

				anim1.setAnimationListener(animationListener);
				anim2.setAnimationListener(animationListener);

				imageView1.startAnimation(anim1);
				imageView2.startAnimation(anim2);
			}
		});
	}

	public static int randomDiceValue() {
		return RANDOM.nextInt(6) + 1;
	}
}

File: SlotMachine/src/main/kotlin/com/ssaurel/slotmachine/MainActivity.kt
package com.ssaurel.slotmachine

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import java.util.*

class MainActivity : AppCompatActivity() {
    private var msg: TextView? = null
    private var img1: ImageView? = null
    private var img2: ImageView? = null
    private var img3: ImageView? = null
    private var wheel1: Wheel? = null
    private var wheel2: Wheel? = null
    private var wheel3: Wheel? = null
    private var btn: Button? = null
    private var isStarted: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        img1 = findViewById<View>(R.id.img1) as ImageView
        img2 = findViewById<View>(R.id.img2) as ImageView
        img3 = findViewById<View>(R.id.img3) as ImageView
        btn = findViewById<View>(R.id.btn) as Button
        msg = findViewById<View>(R.id.msg) as TextView

        btn!!.setOnClickListener {
            if (isStarted) {
                wheel1!!.stopWheel()
                wheel2!!.stopWheel()
                wheel3!!.stopWheel()

                if (wheel1!!.currentIndex == wheel2!!.currentIndex && wheel2!!.currentIndex == wheel3!!.currentIndex) {
                    msg!!.text = "You win the big prize"
                } else if (wheel1!!.currentIndex == wheel2!!.currentIndex || wheel2!!.currentIndex == wheel3!!.currentIndex
                    || wheel1!!.currentIndex == wheel3!!.currentIndex
                ) {
                    msg!!.text = "Little Prize"
                } else {
                    msg!!.text = "You lose"
                }

                btn!!.text = "Start"
                isStarted = false

            } else {

                wheel1 = Wheel(
                    Wheel.WheelListener { img -> runOnUiThread { img1!!.setImageResource(img) } },
                    200,
                    randomLong(0, 200)
                )

                wheel1!!.start()

                wheel2 = Wheel(
                    Wheel.WheelListener { img -> runOnUiThread { img2!!.setImageResource(img) } },
                    200,
                    randomLong(150, 400)
                )

                wheel2!!.start()

                wheel3 = Wheel(
                    Wheel.WheelListener { img -> runOnUiThread { img3!!.setImageResource(img) } },
                    200,
                    randomLong(150, 400)
                )

                wheel3!!.start()

                btn!!.text = "Stop"
                msg!!.text = ""
                isStarted = true
            }
        }
    }

    companion object {

        val RANDOM = Random()

        fun randomLong(lower: Long, upper: Long): Long {
            return lower + (RANDOM.nextDouble() * (upper - lower)).toLong()
        }
    }
}
